core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -246,13 +250,13 @@ private boolean isLoadBalancerEnabled(StackType type, DetailedEnvironmentRespons
 
     private boolean isLoadBalancerEnabledForDatalake(StackType type, DetailedEnvironmentResponse environment) {
         return StackType.DATALAKE.equals(type) && environment != null &&
-            (entitlementService.datalakeLoadBalancerEnabled(ThreadBasedUserCrnProvider.getAccountId()) ||
-                isLoadBalancerEntitlementNotRequiredForCloudProvider(environment.getCloudPlatform()) ||
+            (!isLoadBalancerEntitlementRequiredForCloudProvider(environment.getCloudPlatform()) ||
+                entitlementService.datalakeLoadBalancerEnabled(ThreadBasedUserCrnProvider.getAccountId()) ||
```

Comment:
Whoops sorry, I remember you had said in my previous PR for removing the entitlement that this order was more optimal since it just checks a string first, and then checks the entitlement which is a more expensive operation.
I can revert this if you want.

Reply:
I suspected that may be the reason :). That is generally true, yes, but in this particular case I did it for a reason. You couldn't have known that though, so I appreciate you trying to optimize the code. But yes, please put it back.